src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,1984 @@
+        public int CompareTo(object? obj)
+        {
+            if (obj is not BFloat16 other)
+            {
+                return (obj is null) ? 1 : throw new ArgumentException(SR.Arg_MustBeBFloat16);
+            }
+            return CompareTo(other);
+        }
+
+        /// <summary>
+        /// Compares this object to another object, returning an integer that indicates the relationship.
+        /// </summary>
+        /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
+        public int CompareTo(BFloat16 other) => ((float)this).CompareTo((float)other);
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Equality(TSelf, TOther)" />
+        public static bool operator ==(BFloat16 left, BFloat16 right) => (float)left == (float)right;
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther, TResult}.op_Inequality(TSelf, TOther)" />
+        public static bool operator !=(BFloat16 left, BFloat16 right) => (float)left != (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThan(TSelf, TOther)" />
+        public static bool operator <(BFloat16 left, BFloat16 right) => (float)left < (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_GreaterThan(TSelf, TOther)" />
+        public static bool operator >(BFloat16 left, BFloat16 right) => (float)left > (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_LessThanOrEqual(TSelf, TOther)" />
+        public static bool operator <=(BFloat16 left, BFloat16 right) => (float)left <= (float)right;
+
+        /// <inheritdoc cref="IComparisonOperators{TSelf, TOther, TResult}.op_GreaterThanOrEqual(TSelf, TOther)" />
+        public static bool operator >=(BFloat16 left, BFloat16 right) => (float)left >= (float)right;
+
+        // Equality
+
+        /// <summary>
+        /// Returns a value that indicates whether this instance is equal to a specified <paramref name="other"/> value.
+        /// </summary>
+        public bool Equals(BFloat16 other) => ((float)this).Equals((float)other);
+
+        /// <summary>
+        /// Returns a value that indicates whether this instance is equal to a specified <paramref name="obj"/>.
+        /// </summary>
+        public override bool Equals(object? obj) => obj is BFloat16 other && Equals(other);
+
+        /// <summary>
+        /// Serves as the default hash function.
+        /// </summary>
+        public override int GetHashCode() => ((float)this).GetHashCode();
+
+        /// <summary>
+        /// Returns a string representation of the current value.
+        /// </summary>
+        public override string ToString() => ((float)this).ToString();
```

Comment:
Shouldn't this one also use `Number.FormatBFloat16` rather than deferring to `float`?

Reply:
Yes, but this depends on the refactor to make `Number.FormatFloat` generic math. I'm not sure whether to combine that in this PR.

BTW, I haven't figured out how to calculate the maximum digits to guarantee round-trip, as well as the `PrecisionCustomFormat` parameter.